data-upgrade-news/src/main/java/org/exoplatform/news/upgrade/targets/PublishedNewsDisplayedPropUpgrade.java

```diff
@@ -0,0 +1,162 @@
+                                                                                                          // classes
+
+  public PublishedNewsDisplayedPropUpgrade(InitParams initParams,
+                                           EntityManagerService entityManagerService,
+                                           NewsService newsService,
+                                           MetadataService metadataService,
+                                           PortalContainer container) {
+    super(initParams);
+    this.entityManagerService = entityManagerService;
+    this.newsService = newsService;
+    this.metadataService = metadataService;
+    this.container = container;
+  }
+
+  public int getMigratedPublishedNewsCount() {
+    return migratedPublishedNewsCount;
+  }
+
+  @Override
+  public void processUpgrade(String oldVersion, String newVersion) {
+    long startupTime = System.currentTimeMillis();
+    LOG.info("Start published news migration");
+    List<MetadataItemEntity> newsTargetsMetadataItems = getNewsTargetMetadataItems();
+
+    int totalPublishedNewsCount = newsTargetsMetadataItems.size();
+    LOG.info("Total number of published news to be migrated: {}", totalPublishedNewsCount);
+    int notMigratedPublishedNewsCount = 0;
+    int processedPublishedNewsCount = 0;
+    for (List<MetadataItemEntity> newsTargetsMetadataItemsChunk : ListUtils.partition(newsTargetsMetadataItems, 10)) {
+      int notMigratedPublishedNewsCountByTransaction = manageNewsTargetsMetadataItemsProps(newsTargetsMetadataItemsChunk);
+      int processedPublishedNewsCountByTransaction = newsTargetsMetadataItemsChunk.size();
+      processedPublishedNewsCount += processedPublishedNewsCountByTransaction;
+      migratedPublishedNewsCount += processedPublishedNewsCountByTransaction - notMigratedPublishedNewsCountByTransaction;
+      notMigratedPublishedNewsCount += notMigratedPublishedNewsCountByTransaction;
+      LOG.info("Published news migration progress: processed={}/{} succeeded={} error={}",
+               processedPublishedNewsCount,
+               totalPublishedNewsCount,
+               migratedPublishedNewsCount,
+               notMigratedPublishedNewsCount);
+    }
+    if (notMigratedPublishedNewsCount == 0) {
+      LOG.info("End published news successful migration: total={} succeeded={} error={}. It tooks {} ms.",
+               totalPublishedNewsCount,
+               migratedPublishedNewsCount,
+               notMigratedPublishedNewsCount,
+               (System.currentTimeMillis() - startupTime));
+    } else {
+      LOG.warn("End published news migration with some errors: total={} succeeded={} error={}. It tooks {} ms."
+          + "The not migrated items will be processed again next startup.",
+               totalPublishedNewsCount,
+               migratedPublishedNewsCount,
+               notMigratedPublishedNewsCount,
+               (System.currentTimeMillis() - startupTime));
```

Comment:
```suggestion
               (System.currentTimeMillis() - startupTime));
               throw new IllegalStateException("Some news items wasn't executed successfully. It will be re-attempted next startup");
```